source.go

```diff
@@ -83,20 +107,38 @@ func (s *MultipleSource) Close() error {
 
 func (s *MultipleSource) createSubscribeHandler(prefixSize int, dist *distribute) nats.MsgHandler {
 	return func(msg *nats.Msg) {
+		// call the callback for on subscription received. this will allow users to plugin metrics if they desire.
+		s.opts.onSubscriptionReceived(msg)
+
 		if 0 < prefixSize {
```

Comment:
I know this isn't your code, but refactoring this to reduce the repetition could be useful (reducing the injection points for your extensibility hooks). It appears the only difference is the 1st argument to the `dist.Publish` call is truncated to the `prefixSize` if it's greater than 0.

`ok != true` can also be written as `!ok` - again, not your code 😅

Reply:
Yeah I opted not to change it originally. I think it might be better not too just in case we do want to merge in any updates in the future.

Reply:
Fair call - happy to leave as is.